doc/proposals/rlp-target-gateway-resource.md

```diff
@@ -104,35 +146,76 @@ Proposed new design:
 
 ```yaml
 #  The filter’s behaviour in case the rate limiting service does not respond back. When it is set to true, Envoy will not allow traffic in case of communication failure between rate limiting service and the proxy.
-failureModeDeny: true 
-rateLimitPolicies:
-- name: some_name # policy name
-  rateLimitDomain: some_value # RLS protocol domain value
-  upstreamCluster: rate-limit-cluster # Limitador address reference
-  hostnames: # hostnames
-    - '*.toystore.com'
-  gatewayActions: # rate limit configuration list
-  - rules:
-    - paths: ["/admin/toy"]
-      hosts: ["cars.toystore.com"]
-      methods: ["POST", "DELETE"]
-    configurations:
-    - actions:
-      - generic_key:
-          descriptor_value: yes
-          descriptor_key: admin
+failureModeDeny: true
+rate_limit_policies:
+  - name: toystore
+    rate_limit_domain: toystore-app
+    upstream_cluster: rate-limit-cluster
+    hostnames: ["*.toystore.com"]
+    gateway_actions:
+      - rules:
+          - paths: ["/admin/toy"]
+            methods: ["GET"]
+            hosts: ["pets.toystore.com"]
+        configurations:
+          - actions:
+            - generic_key:
+                descriptor_key: admin
+                descriptor_value: "1"
 ```
 
+Update highlights:
+* [*minor*] `rate_limit_policies` is a list instead of a map indexed by the name/namespace.
+* [*major*] no distinction between "rules" and global actions
+* [*major*] more aligned with RLS: [multiple descriptors](https://www.envoyproxy.io/docs/envoy/latest/api-v3/config/route/v3/route_components.proto#envoy-v3-api-field-config-route-v3-virtualhost-rate-limits) structured by "rate limit configurations" with matching rules
+
 ## WASM-SHIM
 
-Each rate limit policy host has a logical name as well as a set of hostnames to be applied it based on the incoming request’s host header.
+Each rate limit policy has a logical name and a set of hostnames to activate it based on the incoming request’s host header.
+
+The WASM-SHIM builds a tree based data structure holding the rate limit policies.
+The longest (sub)domain match is used to select the policy to be applied.
+Only one policy is being applied per invocation.
```

Comment:
I think we mentioned the usage of two policies for a single request, right?

Reply:
two policies? in what context?

Reply:
> Request 1 (`api.toystore.com`) -> apply RLP A and RLP G
Two policies A and G are being applied, right?

Reply:
Ahhh, make sense now. :+1: